services/shotgrid_common/ayon_shotgrid_hub/update_from_shotgrid.py

```diff
@@ -172,8 +173,9 @@ def create_ay_entity_from_sg_event(
     if not ay_parent_entity:
         # This really should be an edge  ase, since any parent event would
         # happen before this... but hey
-        raise ValueError(
-            "Parent does not exist in Ayon, try doing a Project Sync.")
+        raise MissingParentError(
+            "Parent does not exist in Ayon, this event will be retried"
+            "after while. Hopefully parent will be already created.")
```

Comment:
Missing space + grammar
```suggestion
            "Parent does not exist in Ayon, this event will be retried"
            " after a while. Hopefully parent will be already created.")
```